midi_app_controller/actions/_tests/test_bound_controller.py

```diff
@@ -70,18 +70,24 @@ def bound_controller(binds, controller, actions) -> BoundController:
     return BoundController.create(binds=binds, controller=controller, actions=actions)
 
 
+def validate_actions(data, actions_class):
+    actions = actions_class(**data)
+    x = actions.model_dump()
+
+    for key in x:
+        for key2 in x[key]:
+            if hasattr(data[key], key2):
+                assert getattr(data[key], key2) == x[key][key2]
+
+
 def test_button_actions(actions):
     data = {"action_press": actions[0]}
-    actions = ButtonActions(**data)
-
-    assert actions.dict() == data
+    validate_actions(data, ButtonActions)
```

Comment:
Why not just `actions.model_dump() == data` (where `actions = ButtonActions(**data)`)? It's done this way in `test_utils.py` and `test_controller.py`.

Reply:
data = {'action_press': Action(id='Action1', title='sdfgsdfg', category=None, tooltip=None, status_tip=None, icon=None, icon_visible_in_menu=True, enablement=None, short_title=None, toggled=None, callback=<function actions.<locals>.<lambda> at 0x7a1b7b047e20>, menus=None, keybindings=None, palette=True)}

actions.model_dump() = {'action_press': {'category': None, 'enablement': None, 'icon': None, 'icon_visible_in_menu': True, 'id': 'Action1', 'short_title': None, 'status_tip': None, 'title': 'sdfgsdfg', 'toggled': None, 'tooltip': None}}